pkg/declarativeconfig/transform/auth_provider_test.go

```diff
@@ -0,0 +1,296 @@
+				iap.AudienceConfigKey: "some-audience",
+			},
+		},
+		"oidc config": {
+			authProvider: &declarativeconfig.AuthProvider{
+				OIDCConfig: &declarativeconfig.OIDCConfig{
+					Issuer:                    "some-issuer",
+					CallbackMode:              "auto",
+					ClientID:                  "some-client-id",
+					ClientSecret:              "some-client-secret",
+					DisableOfflineAccessScope: true,
+				},
+			},
+			cfg: map[string]string{
+				oidc.IssuerConfigKey:                    "some-issuer",
+				oidc.ModeConfigKey:                      "auto",
+				oidc.ClientIDConfigKey:                  "some-client-id",
+				oidc.ClientSecretConfigKey:              "some-client-secret",
+				oidc.DisableOfflineAccessScopeConfigKey: "true",
+			},
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			cfg := getConfig(c.authProvider)
+			assert.Equal(t, c.cfg, cfg)
+		})
+	}
+}
+
+func TestTransform(t *testing.T) {
+	// Set everything and the kitchen sink:
+	// - OIDC with all values given.
+	// - multiple required attributes.
+	// - multiple claim mappings.
+	// - multiple groups.
+	authProvider := &declarativeconfig.AuthProvider{
+		Name:            "test-auth-provider",
+		MinimumRoleName: "Analyst",
+		UIEndpoint:      "localhost:8000",
+		Groups: []declarativeconfig.Group{
+			{
+				AttributeKey:   "email",
+				AttributeValue: "someone@something.com",
+				RoleName:       "Admin",
+			},
+			{
+				AttributeKey:   "email",
+				AttributeValue: "somewhere@something.com",
+				RoleName:       "Scope Manager",
+			},
+			{
+				AttributeKey:   "userid",
+				AttributeValue: "12333",
+				RoleName:       "Continous Integration",
+			},
+		},
+		RequiredAttributes: []declarativeconfig.RequiredAttribute{
+			{
+				AttributeKey:   "orgid",
+				AttributeValue: "12345",
+			},
+			{
+				AttributeKey:   "custom_thing",
+				AttributeValue: "some-company",
+			},
+		},
+		ClaimMappings: []declarativeconfig.ClaimMapping{
+			{
+				Path: "some.nested.claim",
+				Name: "custom_thing",
+			},
+			{
+				Path: "another.one",
+				Name: "another_thing",
+			},
+		},
+		OIDCConfig: &declarativeconfig.OIDCConfig{
+			Issuer:                    "http://some-issuer",
+			CallbackMode:              "auto",
+			ClientID:                  "some-client-id",
+			ClientSecret:              "some-client-secret",
+			DisableOfflineAccessScope: true,
+		},
+	}
+	expectedAuthProviderID := declarativeconfig.NewDeclarativeAuthProviderUUID(authProvider.Name).String()
+	expectedConfig := map[string]string{
+		oidc.IssuerConfigKey:                    authProvider.OIDCConfig.Issuer,
+		oidc.ModeConfigKey:                      authProvider.OIDCConfig.CallbackMode,
+		oidc.ClientIDConfigKey:                  authProvider.OIDCConfig.ClientID,
+		oidc.ClientSecretConfigKey:              authProvider.OIDCConfig.ClientSecret,
+		oidc.DisableOfflineAccessScopeConfigKey: strconv.FormatBool(authProvider.OIDCConfig.DisableOfflineAccessScope),
```

Comment:
```suggestion
		oidc.DisableOfflineAccessScopeConfigKey: "true",
```
to test that the config value is correctly transformed